types/block.go

```diff
@@ -741,6 +740,160 @@ func (cs *CommitSig) FromProto(csp tmproto.CommitSig) error {
 
 //-------------------------------------
 
+// ExtendedCommitSig is similar to CommitSig, except it also contains the vote
+// extension and vote extension signature related to this commit.
+type ExtendedCommitSig struct {
+	BlockIDFlag        BlockIDFlag `json:"block_id_flag"`
+	ValidatorAddress   Address     `json:"validator_address"`
+	Timestamp          time.Time   `json:"timestamp"`
+	Signature          []byte      `json:"signature"`
+	VoteExtension      []byte      `json:"extension"`
+	ExtensionSignature []byte      `json:"extension_signature"`
+}
+
+// NewExtendedCommitSigAbsent returns new ExtendedCommitSig with
+// BlockIDFlagAbsent. Other fields are all empty.
+// TODO(sergio): When all UTs passing, try removing the Commit and CommitSig version of new functions
+func NewExtendedCommitSigAbsent() ExtendedCommitSig {
```

Comment:
is there a reason this shouldn't be a pointer? 

is sergio's TODO still valid?

Reply:
I've removed the TODO. Also, `ExtendedCommitSig` isn't a pointer because that's how `CommitSig` has been. Not sure as to the reasons why `CommitSig` isn't a pointer.